runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTranslationContext.java

```diff
@@ -158,4 +144,18 @@ public void setCurrentTransform(AppliedPTransform<?, ?, ?> currentTransform) {
   <T extends PValue> T getOutput(PTransform<?, T> transform) {
     return (T) Iterables.getOnlyElement(currentTransform.getOutputs().values());
   }
+
+  @SuppressWarnings("unchecked")
+  static <T> TypeInformation<WindowedValue<T>> getTypeInfo(PCollection<T> collection) {
+    return getTypeInfo(collection.getCoder(), collection.getWindowingStrategy());
+  }
+
+  @SuppressWarnings("unchecked")
+  static <T> TypeInformation<WindowedValue<T>> getTypeInfo(
```

Comment:
The method seems to be the exact copy of the one in `FlinkStreamingTranslationContext`. Maybe we can have a `TypeInformations` helper class?